kyaml/openapi/Makefile

```diff
@@ -38,6 +38,7 @@ $(MYGOBIN)/kpt:
 	../../hack/install_kpt.sh 0.34.0 $(MYGOBIN)
 
 kubernetesapi/swagger.go: $(MYGOBIN)/go-bindata kubernetesapi/swagger.json
+	./editOpenApiData.sh
```

Comment:
I don't think you should have to change the openapi to insert this, so the script shouldn't be needed at all.

Reply:
What would be the alternative? I was told that we should change the local openapi schema (as opposed to specialized code in kustomize).

Reply:
right, that's what we'd do if we decided the schema were "wrong" (and going to be fixed later), which is one possible interpretation of kubernetes/kubernetes#39188.

I think we now agree that merging should look for a `x-kubernetes-list-map-keys` field and use the list to merge if the list exists, then fall back to the singular field `x-kubernetes-patch-merge-key` and use that.

Upshot - kustomize won't merge ports on name (until the schema changes), so it won't support having multiple names for one `{containerPort,protocol}` tuple. 

Presumably the place to lobby for a schema change is on kubernetes/kubernetes#39188.